src/src/PluginStructs/P145_data_struct.cpp

```diff
@@ -0,0 +1,655 @@
+      116.6020682,  // PARA scaling factor value
+      2.769034857,  // PARB exponent value
+      0.00035,      // CORA
+      0.02718,      // CORB
+      1.39538,      // CORC
+      0.0018,       // CORD
+      -0.003333333, // CORE
+      -0.001923077, // CORF
+      1.130128205,  // CORG
+  },
+    // MQ-2
+  {
+      "MQ-2",       // Name
+      "H2",         // gas
+      p145AlgB,     // preferred/tuned algorithm
+      9.83,         // cleanRatio
+      987.99,       // PARA scaling factor
+      -2.162,       // PARB exponent value
+      0.0,          // CORA
+      0.0,          // CORB
+      0.0,          // CORC
+      0.0,          // CORD
+      0.0,          // CORE
+      0.0,          // CORF
+      0.0,          // CORG
+  },
+  // MQ-3
+  {
+      "MQ-3",       // Name
+      "alcohol",    // gas
+      p145AlgB,     // preferred/tuned algorithm
+      60.0,         // cleanRatio
+      0.3934,       // PARA scaling factor
+      -1.504,       // PARB exponent value
+      0.0,          // CORA
+      0.0,          // CORB
+      0.0,          // CORC
+      0.0,          // CORD
+      0.0,          // CORE
+      0.0,          // CORF
+      0.0,          // CORG
+  },
+  // MQ-4
+  {
+      "MQ-4",       // Name
+      "CH4",        // gas
+      p145AlgB,     // preferred/tuned algorithm
+      4.4,          // cleanRatio
+      1012.7,       // PARA scaling
+      -2.786,       // PARB exponent
+      0.0,          // CORA
+      0.0,          // CORB
+      0.0,          // CORC
+      0.0,          // CORD
+      0.0,          // CORE
+      0.0,          // CORF
+      0.0,          // CORG
+  },
+  // MQ-7
+  {
+      "MQ-7",       // Name
+      "CO",         // gas
+      p145AlgB,     // preferred/tuned algorithm
+      27.5,         // cleanRatio
+      491204,       // PARA scaling
+      -5.826,       // PARB exponent
+      0.0,          // CORA
+      0.0,          // CORB
+      0.0,          // CORC
+      0.0,          // CORD
+      0.0,          // CORE
+      0.0,          // CORF
+      0.0,          // CORG   
+  }
+};
+const int nbrOfTypes = (int)(sizeof(sensorData) / sizeof(struct P145_SENSORDEF));
+
+/*****************************************************************************/
+/*!
+@brief  Perform the calibration algorithm
+@param[in] currentRcal  Value for Rzero determined with current measurement
+@return Void
+@note   Collect the highest Rzero during the calibration period
+        At the end of the period take mean value of current and calculated Rzero
+*/
+/*****************************************************************************/
+void P145_data_struct::calibrate (float currentRcal)
+{
+  unsigned long now = millis();
+  unsigned long time = now - last_cal;
+  float lastRcal = rcal;  // Last calculated Rcal this calibration sequence
+  bool doit = false;
+  
+  if (currentRcal > rcal)
+  {
+    rcal = currentRcal;
+  }
+  doit = (time > P145_CALIBRATION_INTERVAL) && (rcal > 0.0);
```

Comment:
This can be a `const bool` as it is not changed after setting it.

Reply:
Setting things to const is also to prevent bugs.
For example you want to check `==` by due to a typo you type `=`.
When you force yourself to have the const type on the left, you will be prompted by the compiler (or even the intellisense highligher) you made an error.

Just make functions and variables const whenever you can and if the compiler starts complaining you need to double check whether it was an oversight to make it const. But most of the time it is just a bug which is detected before anyone else saw it (apart from the compiler :) )